buildSrc/src/software/aws/toolkits/gradle/resources/ValidateMessages.kt

```diff
@@ -0,0 +1,58 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.gradle.resources
+
+import org.gradle.api.DefaultTask
+import org.gradle.api.GradleException
+import org.gradle.api.logging.Logger
+import org.gradle.api.logging.Logging
+import org.gradle.api.tasks.InputFiles
+import org.gradle.api.tasks.TaskAction
+import org.gradle.language.base.plugins.LifecycleBasePlugin.VERIFICATION_GROUP
+import java.io.File
+
+open class ValidateMessages : DefaultTask() {
+    @InputFiles
+    lateinit var paths: List<File>
+
+    init {
+        group = VERIFICATION_GROUP
+    }
+
+    @TaskAction
+    fun validateMessage() {
+        var hasError = false
+        paths.map {
+            it.absolutePath to it.readLines()
+        }.forEach { (filePath, fileLines) ->
+            fileLines
+                // filter out blank lines and comments
+                .filter { it.isNotBlank() && it.trim().firstOrNull() != '#' }
+                .mapNotNull {
+                    if (it.contains("=")) {
+                        it
+                    } else {
+                        LOG.warn(""""$filePath contains invalid message missing a '=': "$it"""")
+                        null
+                    }
+                }
+                .map { it.split("=").first() }
+                .reduce { item1, item2 ->
+                    if (item1 > item2) {
+                        LOG.error("""$filePath is not sorted:"$item1" > "$item2"""")
+                        hasError = true
+                    }
+
+                    item2
+                }
+            if (hasError) {
+                throw GradleException("$filePath has one or more out of order items!")
+            }
+        }
+    }
+
+    private companion object {
+        val LOG: Logger = Logging.getLogger(ValidateMessages::class.java)
```

Comment:
Oh, dont make loggers this way.

Use the `Task#getLogger()` at the log site

Reply:
Changed